components/screens/browser-api/sign-tx/SignTx.tsx

```diff
@@ -0,0 +1,117 @@
+import { ApiResponse, SignTxPayload } from "@/api/message";
+import { ApiExtensionUtils } from "@/api/extension";
+import { NetworkType } from "@/contexts/SettingsContext.tsx";
+import { IWallet } from "@/lib/wallet/interface";
+import { useBoolean } from "usehooks-ts";
+import useWalletManager from "@/hooks/useWalletManager";
+import useRpcClientStateful from "@/hooks/useRpcClientStateful";
+import { Transaction } from "@/wasm/core/kaspa";
+
+type SignTxProps = {
+  wallet: IWallet;
+  networkId: NetworkType;
```

Comment:
Don't pass in props that can be retrieved from hooks. `useRpcClientStateful`

Reply:
https://github.com/forbole/kastle/pull/22/commits/50bd3ab33454cabdc489dc161f8608d09108f910